src/parse/validator.rs

```diff
@@ -621,7 +621,7 @@ impl<'help, 'app, 'parser> Validator<'help, 'app, 'parser> {
     }
 
     // `incl`: an arg to include in the error even if not used
-    fn missing_required_error(&self, matcher: &ArgMatcher, incl: Option<&Id>) -> ClapResult<()> {
+    fn missing_required_error(&self, matcher: &ArgMatcher, incl: Vec<Id>) -> ClapResult<()> {
```

Comment:
`incl: &[Id]` should be good enough. We don't need the clones and `Vec`

Reply:
If we do need `Vec`, then try to see if `Vec<&Id>` is good enough instead of cloning the `id`.

Reply:
Well it's old behaviour is accepting a reference then clone inside. Using `Vec<&id>` only force us iter it inside and create another vector to hold the temp value, which is even slower. You can check the old code.

Reply:
BTW this is on the error path, performance is not worth worrying.